### PR TITLE
Move CI to Cloudbuild

### PR DESCRIPTION
**Pull request by dimitraz**

### CODE REVIEW

File: .cloudbuild/integration.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - versionName: projects/ravelin-builds/secrets/ci-ngrok-token/versions/latest
+      env: NGROK_AUTH_TOKEN
+
+logsBucket: gs://ravelin-cloudbuild-logs
+options:
+  logging: GCS_ONLY
+
+steps:
+  - id: npm_install
+    name: eu.gcr.io/ravelin-builds/chromium:10.24
```
Comment (icio) [inline .cloudbuild/integration.yaml]: What is this image? Where is it defined? Why are we using an image called "chromium" to run node? Why are we using an image called "chromium" when we're not using chromium?

File: .cloudbuild/integration.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# yaml-language-server: $schema=https://json.schemastore.org/cloudbuild.json
```
Comment (icio) [inline .cloudbuild/integration.yaml]: Any good names for this file that don't overlap with "continuous integration" so much?

**Review by icio — CHANGES_REQUESTED** [review]
Scripts look good but the PR doesn't appear to have triggered the build or received the result, and I'm a bit confused about the appearance of a "chromium" image as we don't appear to have switched to chromedriver for these e2e tests.

File: .cloudbuild/integration.yaml (hunk 2 above)
Comment (williamclot) [reply .cloudbuild/integration.yaml]: `integration_test.yaml`? Open to suggestions.

Comment (icio) [reply .cloudbuild/integration.yaml]: That has the same "integration" overlap. e2e.yaml? core-test.yaml?

File: .cloudbuild/integration.yaml (hunk 1 above)
Comment (williamclot) [reply .cloudbuild/integration.yaml]: You're right we'll only use this image for headless chrome stuff, I've replaced the image. We could use this to run `npm run test:unit` too in the CI build. Up to you if you want to do that or if you consider the browsertack integration tests to be sufficient here.

For your information to answer your question, the image is built here: https://github.com/unravelin/custom-images/blob/main/config.yaml#L2-L9
Dockerfile is here: https://github.com/unravelin/custom-images/blob/main/images/chromium/Dockerfile

**Comment by williamclot** [comment]
I've added some small changes here and there:
- removed some circleCI environment variables and replaced them with cloudbuild equivalents for the buildID to avoid falling back onto using `git`. 
- removed npm `circlewait` dependency which I believe we're not using anymore.
- switched nearly all images to use `node:10.24.1-alpine` instead of the chromium image mentioned previously.
- added an extra build step `npm run test:unit` to run in parallel with `npm run test:integration` which spins up a headless chrome browser using karma.

**Comment by icio** [comment]
> removed npm `circlewait` dependency which I believe we're not using anymore.

Correct, that shouldn't be necessary in cloudbuild. However, I introduced that to ensure only one CI job runs at a time. Otherwise we put too much stress on BrowserStack. Is there an equivalent we can use in cloudbuild?

**Comment by icio** [comment]
The CI build step appears to have run the e2e flow. Was that intended?

**Comment by williamclot** [comment]
The e2e flow was triggered from my local dev not from this PR commit, it's mentioned in this PR because of the github app linking it to the branch and commit ID.

**Comment by williamclot** [comment]
About your comment around waiting, the build isn't triggered on each commit. It's only triggered on commits linked to open PRs. A strategy would be to use draft PRs only open PRs once they are ready for an integration run through browserstack.

Nonetheless if this is big deal I don't think there's anything as nice as circlewait for cloud build but we could do something along the lines of:


```yaml
- id: wait
  name: gcr.io/cloud-builders/gcloud
  timeout: 1200s
  waitFor:
    - npm_run_build
  entrypoint: bash
  args:
    - -c
    - |-
      while :
      do
        builds=$(gcloud builds list --ongoing --format='value(id)' --filter="substitutions.TRIGGER_NAME=$TRIGGER_NAME" --project=$PROJECT_ID | wc -l)
        if [ $$builds -gt "0" ]
        then
          echo "build in progress, waiting..."
          sleep 10
        else
          exit 0
        fi
      done
```

I'm not a big fan of the above, feels quite hacky to me. I would rather keep things as is and revisit this only if really required.

**Comment by icio** [comment]
> It's only triggered on commits linked to open PRs.

Yes but it's still triggered on each commit linked to an open PR. Lets start without it. Not sure that your snippet would work, though. As soon as you've got two queued builds they'd block each other forever.

**Review by icio — APPROVED** [review]
You'll need to update the Github repo's required actions before you'll be able to merge this.

**Pull request merged**